From 1c6276766eb0cb15ebefbf14f81accd3fe7716bc Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Tue, 14 Jan 2014 15:07:43 +0100
Subject: [PATCH 32/40] memory: move private types to exec.c

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <1389712071-23303-33-git-send-email-quintela@redhat.com>
Patchwork-id: 56687
O-Subject: [RHEL7 qemu-kvm PATCH 32/40] memory: move private types to exec.c
Bugzilla: 997559
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert (git) <dgilbert@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 1db8abb10243abe969a2ba307664ba51b60fcac6)
Signed-off-by: Juan Quintela <quintela@trasno.org>
---
 exec.c                         | 16 ++++++++++++++++
 include/exec/memory-internal.h | 15 ---------------
 2 files changed, 16 insertions(+), 15 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 exec.c                         |   16 ++++++++++++++++
 include/exec/memory-internal.h |   15 ---------------
 2 files changed, 16 insertions(+), 15 deletions(-)

diff --git a/exec.c b/exec.c
index 2bf71bf..12adb12 100644
--- a/exec.c
+++ b/exec.c
@@ -82,6 +82,22 @@ int use_icount;
 
 #if !defined(CONFIG_USER_ONLY)
 
+typedef struct PhysPageEntry PhysPageEntry;
+
+struct PhysPageEntry {
+    uint16_t is_leaf : 1;
+     /* index into phys_sections (is_leaf) or phys_map_nodes (!is_leaf) */
+    uint16_t ptr : 15;
+};
+
+struct AddressSpaceDispatch {
+    /* This is a multi-level map on the physical address space.
+     * The bottom level has pointers to MemoryRegionSections.
+     */
+    PhysPageEntry phys_map;
+    MemoryListener listener;
+};
+
 static MemoryRegionSection *phys_sections;
 static unsigned phys_sections_nb, phys_sections_nb_alloc;
 static uint16_t phys_section_unassigned;
diff --git a/include/exec/memory-internal.h b/include/exec/memory-internal.h
index d2b85de..327d54e 100644
--- a/include/exec/memory-internal.h
+++ b/include/exec/memory-internal.h
@@ -22,24 +22,9 @@
 #ifndef CONFIG_USER_ONLY
 #include "hw/xen/xen.h"
 
-typedef struct PhysPageEntry PhysPageEntry;
-
-struct PhysPageEntry {
-    uint16_t is_leaf : 1;
-     /* index into phys_sections (is_leaf) or phys_map_nodes (!is_leaf) */
-    uint16_t ptr : 15;
-};
 
 typedef struct AddressSpaceDispatch AddressSpaceDispatch;
 
-struct AddressSpaceDispatch {
-    /* This is a multi-level map on the physical address space.
-     * The bottom level has pointers to MemoryRegionSections.
-     */
-    PhysPageEntry phys_map;
-    MemoryListener listener;
-};
-
 void address_space_init_dispatch(AddressSpace *as);
 void address_space_destroy_dispatch(AddressSpace *as);
 
-- 
1.7.1

